internal/cssc/cssc.go

```diff
@@ -0,0 +1,190 @@
+		Client: retry.DefaultClient,
+		Cache:  auth.NewCache(),
+		Credential: auth.StaticCredential(loginURL, auth.Credential{
+			Username: username,
+			Password: password,
+		}),
+	}
+
+	// Get manifest and read content
+	_, pulledManifestContent, err := oras.FetchBytes(ctx, repo, filterRepoTagName, oras.DefaultFetchBytesOptions)
+	if err != nil {
+		return Filter{}, errors.Wrap(err, "error fetching manifest content when reading the filter policy")
+	}
+	var pulledManifest v1.Manifest
+	if err := json.Unmarshal(pulledManifestContent, &pulledManifest); err != nil {
+		panic(err)
+	}
+	var fileContent []byte
+	for _, layer := range pulledManifest.Layers {
+		fileContent, err = content.FetchAll(ctx, repo, layer)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Unmarshal the JSON file data to Filter struct
+	var filter = Filter{}
+	if err := json.Unmarshal(fileContent, &filter); err != nil {
+		return Filter{}, errors.Wrap(err, "error unmarshalling json content when reading the filter policy")
+	}
+	return filter, nil
+}
+
+// Reads the filter json from the specified file path and returns the Filter struct
+func GetFilterFromFilePath(filePath string) (Filter, error) {
+	file, err := os.ReadFile(filePath)
+	if err != nil {
+		return Filter{}, errors.Wrap(err, "error reading the filter json file from file path")
+	}
+
+	var filter = Filter{}
+	if err := json.Unmarshal(file, &filter); err != nil {
+		return Filter{}, errors.Wrap(err, "error unmarshalling json content when reading the filter file from file path")
+	}
+
+	return filter, nil
+}
+
+// Applies filter to filter out the repositories and tags from the ACR according to the specified criteria and returns the FilteredRepository struct
+func ApplyFilterAndGetFilteredList(ctx context.Context, acrClient api.AcrCLIClientInterface, filter Filter) ([]FilteredRepository, error) {
+	var filteredRepos []FilteredRepository
+
+	for _, filterRepo := range filter.Repositories {
+		if filterRepo.Enabled != nil && !*filterRepo.Enabled {
+			continue
+		}
+		if filterRepo.Repository == "" || filterRepo.Tags == nil || len(filterRepo.Tags) == 0 || filterRepo.Tags[0] == "" {
+			continue
+		}
+
+		tagList, err := tag.ListTags(ctx, acrClient, filterRepo.Repository)
+		if err != nil {
+			if strings.Contains(err.Error(), "404") {
+				continue
+			}
+			return nil, err
+		}
+
+		if len(filterRepo.Tags) == 1 && filterRepo.Tags[0] == "*" {
+			for _, tag := range tagList {
+				if strings.Contains(*tag.Name, "-patched") {
+					originalTag := strings.Split(*tag.Name, "-patched")[0]
+					matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: originalTag, PatchTag: *tag.Name}
+					filteredRepos = AppendElement(filteredRepos, matchingRepo)
+				} else {
+					matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: *tag.Name, PatchTag: *tag.Name}
+					filteredRepos = AppendElement(filteredRepos, matchingRepo)
+				}
+			}
+		} else {
+			for _, ftag := range filterRepo.Tags {
```

Comment:
The logic here is not that clear to me from the code.  I am not sure clear on what the ftag is expected to be like. Do we have a spec on this for my reference?

Reply:
I'll add some comments in there to make it clearer. Yes, we do have a spec, I'll share it shortly.